src/commands/kol/player.ts

```diff
@@ -0,0 +1,87 @@
+import {
+    ChatInputCommandInteraction,
+    SlashCommandBuilder,
+    bold, 
+    hyperlink
+  } from "discord.js";
+  
+import { kolClient } from "../../clients/kol";
+import { pluralize, toKoldbLink, toMuseumLink, toSnapshotLink } from "../../utils";
+
+
+export const data = new SlashCommandBuilder()
+  .setName("player")
+  .setDescription("Look up information on a given player.")
+  .addStringOption((o) =>
+    o
+      .setName("player")
+      .setDescription("The KOL username or ID of the player you're looking up.")
+      .setRequired(true)
+    );
+
+export async function execute(interaction: ChatInputCommandInteraction) {
+    const member = interaction.member;
+  
+    if (!member) {
+      interaction.reply({
+        content: "You have to perform this action from within a Guild.",
+        ephemeral: true,
+      });
+      return;
+    }
+
+    const playerNameOrId = interaction.options.getString("player", true);
+
+    await interaction.deferReply();
+
+    const player = await kolClient.getPartialPlayer(playerNameOrId);
+
+    if (!player) {
+        interaction.editReply({content: `According to KOL, player ${playerNameOrId} does not exist.`});        
+        return;
+    }
+
+    const playerInfo = await kolClient.getPlayerInformation(player.id);
+
+    if (!playerInfo) {
+        interaction.editReply({content: `While player ${playerNameOrId} exists, this command didn't work. Weird.`});
+        return;    
+    }
+
+    const displayCaseLine = playerInfo.hasDisplayCase ? `Check out this player's display case at ${bold(hyperlink("their museum page!", toMuseumLink(String(player.id))))}` : "";
+
+    const playerInfoFormatted = [
+        `This user is currently a ${bold(player.class)} at level ${player.level}`,
+        `They have completed ${bold(
+            hyperlink(
+                    `${playerInfo.ascensions}`,
+                    toKoldbLink(player.name)))} 
+            ${pluralize(playerInfo.ascensions,"ascension","ascensions")}.`,
+        displayCaseLine,
+        `Loves eating ${playerInfo.favoriteFood}.`,
+        `Loves drinking ${playerInfo.favoriteBooze}.`,
+        ``,
+        `If it exists, their av-snapshot is located ${hyperlink(
+            "here", toSnapshotLink(player.name))}`,
+        ``,
+        `${player.name} last logged in on ${playerInfo.lastLogin}`
+
+    ]
+
+    try {
+        await interaction.editReply({
+            content: null,
+            embeds: [
+                {
```

Comment:
Use embedbuilder please. And add these info bits as fields to the embed rather than content you string together yourself

Reply:
i couldn't really figure out how you were using embedbuilder for the other embeds so instead i looked at the discord.js docs and tried to use it as best as i could understand it. i think this is better than it was before but still probably not what you wanted here, especially with the hacky description piece. let me know if there is better documentation for embedbuilder than [this one from the docs](https://discordjs.guide/popular-topics/embeds.html#embed-preview).

Reply:
So most of the embeds have a "any problems? message docrostov" bit at the bottom. `createEmbed()` kickstarts an embed with that footer but otherwise returns a regular embed builder. One thing I still think about the change you've made is that basically all the data could be in fields (rather than just some), which would be nice.